### PR TITLE
Inject into config.py the methods to document in RefMan

### PR DESCRIPTION
**Pull request by jrobinAV**
WARNING to be merged into release/2.0

### CODE REVIEW

File: tools/setup_generation.py [hunk 1]
```diff
@@ -595,6 +596,56 @@ def generate_entries(entry_infos, package, type, package_output_file, in_group):
...
+from .common.scope import Scope
+from .common.frequency import Frequency
+from taipy.core.common.default_custom_document import DefaultCustomDocument
+from taipy.core.config.job_config import JobConfig
+from taipy.core.config.data_node_config import DataNodeConfig
+from taipy.core.config.task_config import TaskConfig
+from taipy.core.config.scenario_config import ScenarioConfig
+from taipy.core.config.pipeline_config import PipelineConfig\n"""
+    contents.insert(11, imports_to_inject)
+    contents.insert(len(contents) - 2, methods_to_inject)
```
Comment (FabienLelaquais) [inline tools/setup_generation.py:633]: That is pretty dangerous, isn't it? If we change anything in the source file...
A safer (?) way could be to locate the class definition, skip the doc, then fill the body...

**Review by FabienLelaquais — APPROVED** [review]
Nice.
I think I would have created a new step for that... but hey

File: tools/setup_generation.py (hunk 1 above)
Comment (jrobinAV) [reply tools/setup_generation.py:633]: Yes, you are right.
I propose to do what you propose in the develop branch. This is for the 2.0 branch. In 2.0 branch, I believe config.py will not change.

**Comment by jrobinAV** [comment]
> Nice. I think I would have created a new step for that... but hey

Good idea. In develop branch, the steps are well-designed. I will do it using a new step!

File: tools/setup_generation.py (hunk 1 above)
Comment (FabienLelaquais) [reply tools/setup_generation.py:633]: That makes a lot of sense. And if it does, we'll know soon enough.

**Pull request merged**